tests/plugin_tests/chowdsp_plugin_utils_test/GlobalSettingsTest.cpp

```diff
@@ -225,6 +225,65 @@ class GlobalSettingsTest : public TimedUnitTest
         }
     }
 
+    /**
+     * Primarily tests that the ScopedLock's within GlobalPluginSettings are working as they should
+     * Without the ScopedLock's in GlobalPluginSettings this test should have some random failing tests or cause a segmentation fault
+     */
+    void twoInstancesAccessingSameFileTest()
+    {
+        chowdsp::SharedPluginSettings pluginSettings;
+        pluginSettings->initialise (settingsFile, 1);
+
+        std::atomic<bool> thread1Finished { false };
+        juce::Thread::launch (
+            [&] {
+                chowdsp::SharedPluginSettings thread1Settings;
+                thread1Settings->initialise (settingsFile, 1);
+                for (int i = 0; i < 400; ++i)
```

Comment:
I experimented with different values for number of iterations, I found it was quite rare to get errors without the Locks with 300 iterations per thread, but at 400 I was consistently getting errors without the locks which is why I settled on 400 iterations per thread. Let me know if this is too many.

Reply:
Nice! In the CI it looks like the whole Global Settings Test takes ~10 seconds, so I think 400 iterations is totally fine.